### ISSUE TITLE
Streaming semantics (API)

### ISSUE THREAD

**Issue by embano1** [samber/lo#15]
I just learned about this project (been using RxGo), and the API and helpers look really useful.

I mostly work on streaming-based systems/APIs in Go so was wondering if you're also considering adding support for such use cases?

E.g. filter and map based a _stream_ (Go channel?) of `Order` structs coming from a messaging system (Kafka, Rabbit, Redis) or HTTP (gRPC) poll/push API.

**Comment by fsouza** [comment]
Disclaimer: I'm not a maintainer

@embano1 I imagine RxGo will eventually adopt generics?

**Comment by embano1** [comment]
> @embano1 I imagine RxGo will eventually adopt generics?

Yes, this is my understanding, but not sure if the project currently has the bandwidth

**Comment by serge-hulne** [comment]
Here is an alternative to RxGo (with generics) : https://github.com/serge-hulne/go_iter

**Comment by samber** [comment]
@embano1 @serge-hulne @fsouza 

I finally found time to work on this ✌️

I've implemented a dedicated Go package for reactive programming: https://github.com/samber/ro

Further discussion will take place there.

PS: We added support for Go iterators in lo >= 1.52.
